### ISSUE TITLE
Overhaul Installer - migrate to Advanced Installer and split into 32/64 bit MSI's

### ISSUE THREAD

**Issue by Johno-ACSLive** [gitextensions/gitextensions#7368]
## Feature description

WiX is quite demanding to up-keep and therefore simpler MSI creation and management is required. Propose to move to Advanced Installer as it has an intuitive UI/UX and can be driven via command line. Currently, professional edition is required to create an installer with equivalent capability.

During this transition, it is recommended to split the installer and have a separate installer per OS Architecture (1x MSI for 32-bit and 1x MSI for 64-bit).

**Comment by gerhardol** [comment]
A paid license will be a problem
Open source licenses are offered:
https://www.advancedinstaller.com/free-license.html

@RussKie Should we approach Advanced Installer and ask for terms before we go in this direction?
If the installer may run in AppVeyor and run for PRs, it may be OK.

**Comment by Johno-ACSLive** [comment]
It wouldn't hurt to have a chat to them, my experience has been pretty positive with them.

I'm happy to continue with the work in parallel with your discussions. If it turns out the project cannot go with Advanced Installer then that's fine.

Advanced Installer has added quite a few enhancements to command line building over the last couple of versions which is in my list to test. AppVeyor should be able to build the installer just fine.

**Comment by RussKie** [comment]
I dropped them a line.

My biggest question is whether a licence would be tied to a build server, or whether any contributor would be able to build an installer.

**Comment by RussKie** [comment]
FYI, I got a licence, just clarifying the terms of use and how it can be incorporated into our CI/CD.
Will keep you posted.

**Comment by pmiossec** [comment]
>During this transition, it is recommended to split the installer and have a separate installer per OS Architecture (1x MSI for 32-bit and 1x MSI for 64-bit).

I don't know your point of view on that but my feeling is that we could drop the 32-bit support (especially if add some more work).

**Comment by RussKie** [comment]
That is a fresh idea! :D
:+1: from me

**Comment by Johno-ACSLive** [comment]
Do we have telemetry on % of 32-bit Windows installs vs 64-bit? I would think that most, if not all users are on 64-bit Windows installs but I think would be nice to have confirmation.

Agree it would be easier to maintain a 64-bit installer only.

It does also remove the problem of trying to detect OS architecture to pick the correct MSI etc. but creates another in that 32-bit Windows installs will fail MSI installation via the updater. The user will still have the previous version installed but if they try update it will just fail to update.

This can be of course mitigated by updating the Wiki to mention we are only supporting 64-bit Windows installations via the installer and users should switch to the portable if they still need the 32-bit binaries.

**Comment by pmiossec** [comment]
I think that we could add it later if a lot of people ask for it.

I look for statistics but found only one source which is steam. I don't know if it's representative but only around 1% of users use a 32bit.

You could also find companies stopping supporting 32bits. I think that, as a open source project, we should not target something even companies with resources don't try to do.

**Comment by gerhardol** [comment]
> I look for statistics but found only one source which is steam. 

I expect GE to have many more 32bit usersthan Steam, as gaming PCs are generally more powerful. But portable is a workaround.

> It does also remove the problem of trying to detect OS architecture to pick the correct MSI etc

It will still help to have a best-effort 32bit detection

**Comment by RussKie** [comment]
I checked the AppInsights we have, we don't collect OS bitness. Nor 100% of our users enable AI, so the results we be unreliable anyway.
However I would think that the majority of our users would be on x64. 

If we get users cry loud about x86 installer we can consider adding it later.

**Comment by RussKie** [comment]
We have a licence which we can use on the CI, I'll share the details in the team's chat

To get started:
> To get Advanced installer integrated into your build pipeline you could try to make use of our command line support.
https://www.advancedinstaller.com/user-guide/command-line.html
